### PR TITLE
ICDS Dashboard: SDD detailing changes

### PR DESCRIPTION
**Pull request by saiprakash-v**
https://dimagi-dev.atlassian.net/browse/QA-1357
https://dimagi-dev.atlassian.net/browse/QA-1387

This PR should be **merged only after** https://github.com/dimagi/commcare-hq/pull/27885

### CODE REVIEW

File: custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js [hunk 1]
```diff
@@ -626,12 +626,23 @@ function ServiceDeliveryDashboardController($rootScope, $scope, $http, $location
 
     }
 
+    function isAllRow(rowData) {
+        return (rowData['state_name'] === 'All' && rowData['district_name'] === 'All' && rowData['block_name'] === 'All'
```
Comment (Rohit25negi) [inline custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js:630]: When will this condition be satisfied when all of them will be 'All'?

**Review by Rohit25negi — CHANGES_REQUESTED** [review]
Left one question. but not sure what is intended here

File: custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js (hunk 1 above)
Comment (saiprakash-v) [reply custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js:630]: yes. For "All" row backend is sending all the location names as "All"

**Comment by saiprakash-v** [comment]
> Left one question. but not sure what is intended here

All the values in "VHSND" and "CBE" columns are of type "boolean". But for "All" row in these 2 columns has to display values of other datatype. So I'm checking for the conditions of sector level and "all" row to show proper values

**Comment by saiprakash-v** [comment]
@Rohit25negi I'm working on this ticket (https://dimagi-dev.atlassian.net/browse/QA-1390). Don't merge this PR yet. I will push the changes of this ticket to same PR

**Comment by saiprakash-v** [comment]
> I'm working on this ticket (https://dimagi-dev.atlassian.net/browse/QA-1390). Don't merge this PR yet. I will push the changes of this ticket to same PR

@Rohit25negi changes mentioned in this ticket are done. Gentle reminder to review

File: custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js (hunk 1 above)
Comment (Rohit25negi) [reply custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js:632]: I see

**Review by Rohit25negi — APPROVED** [review]

**Review by Rohit25negi — APPROVED** [review]

**Pull request merged**